src/composables/useCopyToClipboard.ts

```diff
@@ -4,32 +4,68 @@ import { useToast } from 'primevue/usetoast'
 import { t } from '@/i18n'
 
 export function useCopyToClipboard() {
-  const { copy, isSupported } = useClipboard()
+  const { copy, copied, isSupported } = useClipboard()
   const toast = useToast()
 
-  const copyToClipboard = async (text: string) => {
-    if (isSupported) {
-      try {
-        await copy(text)
+  function fallbackCopy(text: string) {
+    const textarea = document.createElement('textarea')
+    textarea.value = text
+    textarea.style.position = 'fixed'
+    textarea.style.opacity = '0'
+    document.body.appendChild(textarea)
+    textarea.select()
+
+    try {
+      const successful = document.execCommand('copy')
+      if (successful) {
         toast.add({
           severity: 'success',
           summary: t('g.success'),
           detail: t('clipboard.successMessage'),
           life: 3000
         })
-      } catch (err) {
+      } else {
         toast.add({
           severity: 'error',
           summary: t('g.error'),
           detail: t('clipboard.errorMessage')
         })
       }
-    } else {
+    } catch (err) {
       toast.add({
         severity: 'error',
         summary: t('g.error'),
-        detail: t('clipboard.errorNotSupported')
+        detail: t('clipboard.errorMessage')
       })
+    } finally {
+      document.body.removeChild(textarea)
+    }
+  }
+
+  const copyToClipboard = async (text: string) => {
+    if (isSupported) {
+      try {
+        await copy(text)
+
+        // Check if copy was successful
+        if (copied.value) {
```

Comment:
To avoid duplication can we do the toast in a finally block and key off the success for failure of the try catch.

That way we only have two toast  branches (success and error) instead of four.